www/updateUes.php

```diff
@@ -36,91 +57,217 @@
+            CLASS_PREFIX . 'There is no UserExtSource that could be used for user ' . $perunUserId . ' and extSourcewith ExtSource ' . $attributesFromIdP[SOURCE_IDP_ENTITY_ID][0]
         );
     }
+} catch (\Exception $ex) {
+    Logger::warning(
+        CLASS_PREFIX . 'Updating UES for user with userId: ' . $perunUserId . ' was not successful: ' .
+        $ex->getMessage()
+    );
+}
+
+/**
+ * @param mixed $adapter
+ * @param mixed $extSourceName
+ * @param mixed $extLogin
+ * @param mixed $attrMap
+ * @param mixed $attrsToConversion
+ * @param mixed $perunUserId
+ * @param mixed $attributesFromIdP
+ */
+function doUpdate(
+    $adapter,
+    $extSourceName,
+    $extLogin,
+    $attrMap,
+    $attrsToConversion,
+    $perunUserId,
+    $attributesFromIdP
+): bool {
+    $userExtSource = getUserExtSource($adapter, $extSourceName, $extLogin);
+    if (null === $userExtSource) {
+        return false;
+    }
+
+    $attrsFromPerun = getAttributesFromPerun($adapter, $attrMap, $userExtSource);
+    if (empty($attrsFromPerun)) {
+        return false;
+    }
+
+    $attrsToUpdate = getAttributesToUpdate($attrsFromPerun, $attrsToConversion, $attributesFromIdP);
+    if (empty($attrsToUpdate)) {
+        return false;
+    }
+
+    return updateUserExtSource($adapter, $attrsToUpdate, $userExtSource, $perunUserId);
+}
+
+function getUserExtSource($adapter, $extSourceName, $extLogin)
+{
+    try {
+        return $adapter->getUserExtSource($extSourceName, $extLogin);
+    } catch (SimpleSAML\Module\perun\Exception $ex) {
+        Logger::debug(CLASS_PREFIX . 'Caught exception when fetching user ext source, probably does not exist.');
+        Logger::debug(CLASS_PREFIX . $ex->getMessage());
+
+        return null;
+    }
+}
 
-    $attributesFromPerunRaw = $adapter->getUserExtSourceAttributes($userExtSource['id'], array_keys($attrMap));
+function getAttributesFromPerun($adapter, $attrMap, $userExtSource): array
+{
+    $attributesFromPerunRaw = $adapter->getUserExtSourceAttributes($userExtSource[ID], array_keys($attrMap));
     $attributesFromPerun = [];
+
     foreach ($attributesFromPerunRaw as $attributeFromPerunRaw) {
-        $attributesFromPerun[$attributeFromPerunRaw['name']] = $attributeFromPerunRaw;
+        $attributesFromPerun[$attributeFromPerunRaw[NAME]] = $attributeFromPerunRaw;
     }
 
     if (null === $attributesFromPerun) {
-        throw new Exception('perun/www/updateUes.php: getting attributes was not successful.');
+        throw new Exception(CLASS_PREFIX . 'Getting attributes was not successful.');
     }
 
+    return $attributesFromPerun;
+}
+
+function getAttributesToUpdate($attributesFromPerun, $attrsToConversion, $attributesFromIdP): array
+{
     $attributesToUpdate = [];
 
     foreach ($attributesFromPerun as $attribute) {
-        $attrName = $attribute['name'];
+        $attrName = $attribute[NAME];
 
         if (isset($attrMap[$attrName], $attributesFromIdP[$attrMap[$attrName]])) {
```

Comment:
Extract the things you want to check in separate variables, it will be much easier to get the code meaning.
I.e. `$mappedAttributeName = $attrMap[$attrName]; $idpAttribute = $attributesFromIdP[$attrMap[$attrName]]`. Also consider introducing checks if the keys exist in the array.